jenkins/hourly_maintenance.py

```diff
@@ -194,6 +196,29 @@ def DeleteCorruptGitRepos():
     return err
 
 
+def DeleteOldWorkspaces():
+    err = 0
+    live_jobs = None
+    for host in ('jenkins-master', 'pull-jenkins-master'):
+        try:
+            resp = requests.get("http://%s:8080/api/json?pretty=true&tree=jobs[name]" % host)
+            resp.raise_for_status()
+            live_jobs = {job['name'] for job in resp.json()['jobs']}
+        except requests.exceptions.ConnectionError:
+            continue
+        except requests.exceptions.RequestException:
+            traceback.print_exc()
+    if live_jobs is None:
+        print 'unable to determine live jenkins jobs, not deleting any workspaces'
+        return 1
+    for dirname in sorted(glob.glob('/var/lib/jenkins/workspace/*')):
+        key = os.path.basename(dirname).replace('@tmp', '')
+        if key not in live_jobs:
```

Comment:
oh cool!

I kind of like the idea of deleting the workspace entirely if it hasn't run on this node for awhile.

Happy to merge this as is btw, definitely better than our present situation